### PR TITLE
begin the process of removing nightwatch and configs from vets-website

### PR DESCRIPTION
**Pull request by CBonade**
## Description
As we started the process of removing Nightwatch from `vets-website`, we discovered a number of specs that had not yet been converted to Cypress. To avoid losing any test coverage, these specs were converted first and the Nightwatch configs were added back so they can still be added.

## Testing done
One hiccup came up when adding everything back - the nightwatch script that runs via yarn to kickoff the test suite was giving me a permission denied error so I had to chmod it in the workflow.  This didn't exist before removal, however nothing showed a reason why it wouldn't work without it anymore.  Files changed on this PR only show test specs and helpers.  Have verified that it runs normally with the `chmod` added.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by cvalarida — APPROVED** [review]
The forms system tests look fine to me (granted, I only skimmed). :+1: 

In the future, I'd recommend splitting changes like this into multiple PRs. It's a lot to review and a lot to ignore for a lot of people.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
What is the purpose of having an empty 01.feedback-tool-keyboard.cypress.spec.js?

**Comment by CBonade** [comment]
> What is the purpose of having an empty 01.feedback-tool-keyboard.cypress.spec.js?

Good catch here - There were two keyboard specs that I started working out but later found out a different team member had been working on keyboard test functions for Cypress for a while so I stopped awaiting that solution being finished (those will come in a separate branch).  Removing this file from this branch.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by k80bowman — APPROVED** [review]

File: src/platform/site-wide/user-nav/tests/user-nav.cypress.spec.js
Comment (AGarcia-Clarity) [inline src/platform/site-wide/user-nav/tests/user-nav.cypress.spec.js:20]: If this is commented out, should it be removed?

**Review by asg5704 — APPROVED** [review]
Comment added. LGTM.

File: src/platform/site-wide/user-nav/tests/user-nav.cypress.spec.js
Comment (CBonade) [reply src/platform/site-wide/user-nav/tests/user-nav.cypress.spec.js:20]: There seems to be a problem with cross-origin testing in Cypress that did not exist in Nightwatch.  I'm currently attempting to come up with a different way to test this button accurately, however, due to the volume of this branch I didn't want to hold up the other tests moving forward in the interim (this will be solved before it's Nightwatch counterpart is removed).

Comment (AGarcia-Clarity) [reply src/platform/site-wide/user-nav/tests/user-nav.cypress.spec.js:20]: Gotcha 👍

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (kelsonic) [inline src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:107]: How come we need to provide a Timeouts option for each of these `.get`? In theory I think Cypress will wait for long enough before failing the test 🙂

File: src/platform/forms/tests/fixtures/mocks/mockUser.js
Comment (kelsonic) [inline src/platform/forms/tests/fixtures/mocks/mockUser.js:31]: Just confirming it's ok if these values are hardcoded and not derived via `moment` like the `expires_at` 🙂

File: src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js
Comment (kelsonic) [inline src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js:116]: Same thing as a previous comment, but I believe `cy.get` defaults to `Timeouts.normal` 🙂

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:107]: Cypress is supposed to do that as you described.  Unfortunately, it seems in GHA we've run into some race conditions that prevent this from happening with the timing that it should. It's still something we're constantly tinkering with and trying to figure out exactly how much of this is needed, but erring on the side of caution to get GHA stable.

File: src/platform/forms/tests/fixtures/mocks/mockUser.js
Comment (CBonade) [reply src/platform/forms/tests/fixtures/mocks/mockUser.js:31]: At present, our objective is more of a "lift and shift" to get us away from Nightwatch and having to run two testing frameworks side by side at deployment. This was how it was being done in Nightwatch so I've copied it over here.  Happy to explore reevaluating in a future initiative

File: yarn.lock
Comment (kelsonic) [inline yarn.lock:2759]: Is it necessary to commit `yarn.lock` changes? 🙂

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 This must have been a lot of work, thank you! When will we remove the respective Nightwatch tests? Exciting!

File: src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js
Comment (Rleahy22) [inline src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js:28]: I'd like to see this as separate expectations, but this is okay

**Review by Rleahy22 — APPROVED** [review]
🚀

File: src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js
Comment (CBonade) [reply src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js:116]: I believe our timeouts helper that's being used has `2000` set as the normal presently, where Cypress defaults to a 4000.  In this case, it's shortening that period. I'm not 100% sure it's necessary, as you've stated, however it's another case of just matching what was stable in Nightwatch.

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:21]: yep, good catch!  Will remove these.

**Review by gescobardev — APPROVED** [review]

File: src/applications/static-pages/tests/application-status.cypress.spec.js
Comment (CBonade) [reply src/applications/static-pages/tests/application-status.cypress.spec.js:87]: Yeah, it probably should.  Moving it to it's own file.

File: src/applications/veteran-representative/tests/00.veteran-representative-entry.cypress.spec.js
Comment (CBonade) [reply src/applications/veteran-representative/tests/00.veteran-representative-entry.cypress.spec.js:20]: Removing this entirely.  That function is not necessary in Cypress that we've seen (was a carry-over relic from Nightwatch)

File: src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js
Comment (CBonade) [reply src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js]: image

Done!

File: yarn.lock
Comment (CBonade) [reply yarn.lock]: I've deleted yarn lock locally and reinstalled with the most recent master merged in.  Not sure why these changes are here nor why they would be flagged as new, as nothing has been added on this branch.

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (mpelzsherman) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:104]: I hope this won't be neccessary for _all_ of our Cypress tests in order to convert to GHA! 😨

File: src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js
Comment (mpelzsherman) [inline src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js:74]: This test appears to be specific to the Facility Locator. Any reason not to name and locate it accordingly?

Comment (CBonade) [reply src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js:74]: Our objective was a strict `lift and shift` of existing Nightwatch structure and just convert in-place to Cypress, would be the only reason I can give for why it was put where it as (as that was it's prior location in Nightwatch).  I am more than happy to move it to the facility locator application if that's considered a better choice.

File: src/platform/site-wide/banners/index.cypress.spec.js
Comment (mpelzsherman) [inline src/platform/site-wide/banners/index.cypress.spec.js:11]: Is this even a valid test? I checked the home page, and those selectors do not work.

Are we planning to fix the test?

**Comment by mpelzsherman** [comment]
I see changes to the `yarn.lock` file, not no changes to `package.json`. This usually indicates a problem. Please make sure any changes to `yarn.lock` are intentional and correct, thanks.

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (mpelzsherman) [inline src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:12]: `{ timeout: Timeouts.normal } `should be the default, no?

**Review by mpelzsherman — CHANGES_REQUESTED** [review]
I'd like to see my comments addressed before this is merged. Thanks!

File: src/platform/site-wide/banners/index.cypress.spec.js
Comment (CBonade) [reply src/platform/site-wide/banners/index.cypress.spec.js:11]: I'm honestly not sure what the plans are for the test. It was being disabled in Nightwatch (assuming because it was either outdated or non-functional) however it was not ever removed or fixed, so it's been copied over in it's current state just to ensure we do not lose it in the event it was a WIP when Nightwatch is removed.  Once the conversion has been made, we'll be reevaluating the suite as a whole and communicating with any necessary parties about tests that may no longer be necessary (as in something like this, that very well may be covered by other accessibility checks) or tests that are not currently being utilized and what their future plans are.

File: src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js
Comment (mpelzsherman) [reply src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js:74]: It is a facility locator test, and should therefore live in `src/applications/facility-locator/tests/e2e`.

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:12]: I'm more than happy to reevaluate this as it's something that was carried over from being in place on Nightwatch (and exists in many other Cypress tests as well). The Cypress default per their docs is `4000` and our default `normal` is `2000` per our helpers file, so we're applying a shorter timeout than the default in this case.  If we feel the default timeout will cover all these cases I'm happy to remove it wherever it's unnecessary.

File: yarn.lock
Comment (kelsonic) [reply yarn.lock]: You can remove the yarn.lock changes by doing `git reset --soft HEAD~1` and then re-committing all the files except the yarn.lock 🙂 Assuming that the yarn.lock was updated in the most recent commit on your branch actually

File: src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js
Comment (kelsonic) [reply src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js:116]: Understood! I do think you can omit the `, { timeout: Timeouts.normal }` for each `cy.get` in this case, but I also totally get wanting to achieve parity at first to make sure nothing breaks 👍

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (kelsonic) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:104]: It's not 🙂 If you feel comfortable with it @CBonade you can try omitting the `, { timeout: Timeouts.slow/normal }` bit from each `cy.get` and if the tests still pass then we should be golden 🏅

Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:104]: It definitely seems situational, and more sensitive immediately following a navigation and/or with inputs vs other elements

File: src/platform/site-wide/banners/index.cypress.spec.js
Comment (mpelzsherman) [reply src/platform/site-wide/banners/index.cypress.spec.js:11]: I understand that the intention here is a "lift and shift", but I see no reason to port a test that adds no value.

**Comment by CBonade** [comment]
> I see changes to the `yarn.lock` file, not no changes to `package.json`. This usually indicates a problem. Please make sure any changes to `yarn.lock` are intentional and correct, thanks.

I've gone ahead and sourced `yarn.lock` off the current master branch, which seems to have done the trick and removed it from proposed changes

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (mpelzsherman) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:12]: Personally I'd rather just use the Cypress default. If we want to redefine the default for all tests, we should do that in our Cypress config, rather than adding an override to every `get`.

File: src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js
Comment (CBonade) [reply src/platform/monitoring/DowntimeNotification/tests/DowntimeNotification.cypress.spec.js]: have reloacted this as requested

File: yarn.lock
Comment (CBonade) [reply yarn.lock]: I was able to re-source yarn lock off the master branch and that seemed to do the trick. `git checkout origin/master -- yarn.lock`

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (U-DON) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js:12]: I agree, we should use the Cypress defaults unless there's a particular reason not to. The timeouts from Nightwatch aren't going to be as relevant with the removal of Nightwatch itself, so I would recommend redefining any nececssary custom timeouts around the Cypress defaults instead.

File: src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js
Comment (CBonade) [reply src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js:116]: Giving a shot at removing these and running the full suite to make sure it's still stable.  Will post in the other conversation regarding this when I have an update!  Resolving this one to condense.

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js]: Thanks for the feedback from you both.  I went ahead and removed any instances of `Timeout: normal` from Cypress get functions across the code base.  I'll run it in CI a few times to be sure that nothing has been unintentionally impacted here.  As many other files were affected, this will require a few other team reviews to go through, but we'll have much cleaner code on the other side of it!

**Comment by mpelzsherman** [comment]
I appreciate you doing this. I think it’s important not to set a bad precedent for a future tests.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js
Comment (CBonade) [reply src/applications/edu-benefits/tests/feedback-tool/00.feedback-tool.cypress.spec.js]: Done (details in other thread)

File: src/platform/site-wide/banners/index.cypress.spec.js
Comment (CBonade) [reply src/platform/site-wide/banners/index.cypress.spec.js:11]: Totally agree with the sentiment here.  My only concern was with the permanent removal of nightwatch that someone may lose something they intended to come back to.  Ran this specific spec by my team and they agreed that we can probably drop this as well, so removing this one.

**Comment by jbalboni** [comment]
It looks like my team was tagged due to formatting changes that appear to be unintentional (they're failing the lint step). I'm assuming that you won't need my team's approval once those changes are fixed.

File: src/applications/pre-need/tests/00-all-fields.cypress.spec.js
Comment (Mottie) [inline src/applications/pre-need/tests/00-all-fields.cypress.spec.js:45]: You're removing all `Timeouts.normal`, right?

File: src/applications/pre-need/tests/00-all-fields.cypress.spec.js
Comment (Mottie) [inline src/applications/pre-need/tests/00-all-fields.cypress.spec.js:72]: Here too

File: src/applications/pre-need/tests/00-all-fields.cypress.spec.js
Comment (Mottie) [inline src/applications/pre-need/tests/00-all-fields.cypress.spec.js:90]: 👀 ... and more below

**Review by mpelzsherman — APPROVED** [review]
Thank you for addressing my concerns so quickly! Much appreciated!

**Comment by CBonade** [comment]
> It looks like my team was tagged due to formatting changes that appear to be unintentional (they're failing the lint step). I'm assuming that you won't need my team's approval once those changes are fixed.

@jbalboni I've removed the files that I believe were the ones you were referencing, however, it's still stating a review is required from your team.  Is there something I can do to rectify this further?

**Comment by CBonade** [comment]
> Thank you for addressing my concerns so quickly! Much appreciated!

@mpelzsherman Thank you for such thorough feedback. I'm always looking for ways I can be more efficient and improve.  Every one of these helps!

**Review by Mottie — APPROVED** [review]

File: src/applications/pre-need/tests/00-all-fields.cypress.spec.js
Comment (CBonade) [reply src/applications/pre-need/tests/00-all-fields.cypress.spec.js:45]: Looks like the formatting outdid my site-wide search for these.  I'll do another pass :)

**Review by jbalboni — APPROVED** [review]
LGTM

**Comment by jbalboni** [comment]
@CBonade Looks like there were some timeout removals in the vaos code that I didn't see before.

**Review by alexandec — APPROVED** [review]

**Comment by CBonade** [comment]
@department-of-veterans-affairs/chatbot-admin , @department-of-veterans-affairs/vsa-authd-exp-frontend , @department-of-veterans-affairs/vsa-bam-2-frontend, @department-of-veterans-affairs/vsa-caregiver-frontend , @department-of-veterans-affairs/vsa-ebenefits-frontend  just tagging here for visibility, if you might please have time to look this over.

**Review by micahchiang — APPROVED** [review]

**Review by tmitchellgcio — APPROVED** [review]

**Review by mdewey — APPROVED** [review]
LGTM

**Review by jgacek — APPROVED** [review]

**Review by craigwheeler — APPROVED** [review]

**Review by ddzz — APPROVED** [review]

**Pull request merged**
